test/unified_format.py

```diff
@@ -1800,7 +1800,8 @@ def format_logs(log_list):
                 clientid = self.entity_map[client["client"]]._topology_settings._topology_id
                 actual_logs = formatted_logs[clientid]
                 actual_logs = [log for log in actual_logs if log["component"] in components]
-                self.assertEqual(len(client["messages"]), len(actual_logs))
+                if not client.get("ignoreExtraMessages", False):
+                    self.assertEqual(len(client["messages"]), len(actual_logs))
```

Comment:
When ignoreExtraMessages is false this new code will mistakenly pass when the test expects messages `[A, B, C]` but actually only sees `[A, B]`. We should handle ignoreExtraMessages by truncating the extras like we do for ignoreExtraEvents:
```python
            if ignore_extra_events:
                actual_events = actual_events[: len(events)]
...
```

Reply:
Fixed